src/components/AssetSearch/AssetList.tsx

```diff
@@ -41,6 +48,33 @@ export const AssetList = ({ assets, handleClick }: AssetListProps) => {
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [assets])
 
+  /**
+   * This function enrich the Asset by adding the user cryptoAmount and marketCap for each asset for facilitate sorting.
+   * @param assets Asset[]
+   * @returns
+   */
+  const enrichAsset = (assets: Asset[]): Asset[] => {
+    return assets.map(asset => {
+      const amount = rowData.find(d => d.assetId === asset.assetId)?.cryptoAmount
+      const assetMarketData = marketData.crypto.byId[asset.assetId]
+      return {
+        ...asset,
+        cryptoAmount: amount ? Number(amount) : 0,
+        marketCap: assetMarketData ? Number(assetMarketData.marketCap) : 0,
+      } as Asset
+    })
+  }
+
```

Comment:
There is a `helpers` search in the same `src/components/AssetSearch/` folder, let's move this there